OpenRA.Mods.Common/Traits/World/ResourceLayer.cs

```diff
@@ -237,7 +149,8 @@ public void AddResource(ResourceType t, CPos p, int n)
 			cell.Density = Math.Min(cell.Type.Info.MaxDensity, cell.Density + n);
 			Content[p] = cell;
 
-			dirty.Add(p);
+			if (CellChanged != null)
+				CellChanged(p);
```

Comment:
... and change all 3 calls to `CellChanged(cell, Content[cell].Type);`  (`p` instead of `cell` in this particular case) ...

> Review 2